don't crash on undefined opts in addNewKeyring

cg505 opened the pull request:
This fixes the "Cannot read property 'password' of undefined" errors when using hardware wallets.

cg505 committed:
don't crash on undefined opts in addNewKeyring

diracdeltas commented on index.js:
```diff
@@ -194,7 +194,7 @@ class KeyringController extends EventEmitter {
   // and this is used to retrieve them from the keyringTypes array.
   async addNewKeyring (type, opts) {
-    if (opts.password) {
+    if (opts && opts.password) {
```
mind adding a quick test for this in test/index.js?

cg505 replied:
@diracdeltas done. It was a little more challenging than expected because the two keyrings that are included by default actually need a certain things passed into `opts`. I created a mock keyring that doesn't need `opts`, and just for the one test, a new `KeyringController` that includes that keyring type.

cg505 committed:
add test for adding keyring with no opts

We need to create a mock keyring that does not require `opts` to be
passed when constructing, because both the SimpleKeyring and HdKeyring
require `opts` or they will fail.

ryanml reviewed (approved):
lgtm, will hold for further review
